py-rattler/tests/unit/test_index.py

```diff
@@ -30,23 +39,109 @@ def test_index(package_directory):
         assert "pytweening-1.0.4-pyhd8ed1ab_0" in f.read()
 
 
-def test_index_specific_subdir_non_noarch(package_directory):
-    index(package_directory, Platform("win-64"))
+@pytest.mark.asyncio
+async def test_index_specific_subdir_non_noarch(package_directory):
+    await index_fs(package_directory, Platform("win-64"))
 
     assert "repodata.json" in os.listdir(package_directory / "win-64")
     with open(package_directory / "win-64/repodata.json") as f:
         assert "ruff-0.0.171-py310h298983d_0" in f.read()
-    assert "repodata.json" in os.listdir(package_directory / "noarch")
-    with open(package_directory / "noarch/repodata.json") as f:
-        assert "pytweening-1.0.4-pyhd8ed1ab_0" in f.read()
 
 
-def test_index_specific_subdir_noarch(package_directory):
-    index(package_directory, Platform("noarch"))
+@pytest.mark.asyncio
+async def test_index_specific_subdir_noarch(package_directory):
+    await index_fs(package_directory, Platform("noarch"))
 
     win_files = os.listdir(package_directory / "win-64")
     assert "repodata.json" not in win_files
     assert "ruff-0.0.171-py310h298983d_0.conda" in win_files
     assert "repodata.json" in os.listdir(package_directory / "noarch")
     with open(package_directory / "noarch/repodata.json") as f:
         assert "pytweening-1.0.4-pyhd8ed1ab_0" in f.read()
+
+
+# ---------------------------------------- S3 ---------------------------------------- #
+
+
+@dataclass
+class S3Config:
+    access_key_id: str
+    secret_access_key: str
+    region: str = "auto"
+    endpoint_url: str = "https://e1a7cde76f1780ec06bac859036dbaf7.r2.cloudflarestorage.com"
+    bucket_name: str = "rattler-build-upload-test"
+    channel_name: str = field(default_factory=lambda: f"channel{uuid.uuid4()}")
+
+
+@pytest.fixture()
+def s3_config() -> S3Config:
+    access_key_id = os.environ.get("RATTLER_TEST_R2_ACCESS_KEY_ID")
+    if not access_key_id:
+        pytest.skip("RATTLER_TEST_R2_ACCESS_KEY_ID environment variable is not set")
+    secret_access_key = os.environ.get("RATTLER_TEST_R2_SECRET_ACCESS_KEY")
+    if not secret_access_key:
+        pytest.skip("RATTLER_TEST_R2_SECRET_ACCESS_KEY environment variable is not set")
```

Comment:
@delsner these secrets need to be present in the github workflow as env vars, @wolfv could you add them as repo secrets?